OpenSim/Simulation/Model/OffsetFrame.h

```diff
@@ -0,0 +1,159 @@
+#ifndef OPENSIM_OFFSET_FRAME_H_
+#define OPENSIM_OFFSET_FRAME_H_
+/* -------------------------------------------------------------------------- *
+ *                          OpenSim:  OffsetFrame.h                           
+ * -------------------------------------------------------------------------- *
+ * The OpenSim API is a toolkit for musculoskeletal modeling and simulation.  *
+ * See http://opensim.stanford.edu and the NOTICE file for more information.  *
+ * OpenSim is developed at Stanford University and supported by the US        
+ * National Institutes of Health (U54 GM072970, R24 HD065690) and by DARPA    
+ * through the Warrior Web program.                                           
+ *                                                                            
+ * Copyright (c) 2005-2013 Stanford University and the Authors                
+ * Author(s): Matt DeMers Ajay Seth, Ayman Habib                              
+ *                                                                            
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may    
+ * not use this file except in compliance with the License. You may obtain a  *
+ * copy of the License at http://www.apache.org/licenses/LICENSE-2.0.         
+ *                                                                            
+ * Unless required by applicable law or agreed to in writing, software        
+ * distributed under the License is distributed on an "AS IS" BASIS,          
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.   *
+ * See the License for the specific language governing permissions and        
+ * limitations under the License.                                             
+ * -------------------------------------------------------------------------- */
+
+// INCLUDE
+#include <OpenSim/Simulation/osimSimulationDLL.h>
+#include <OpenSim/Simulation/Model/Frame.h>
+#include <OpenSim/Simulation/Model/PhysicalFrame.h>
+
+namespace OpenSim {
+
+//=============================================================================
+//=============================================================================
+/**
+ * An OffsetFrame is a Frame whose transform (translation and orientation)
+ * with respect to another (parent) Frame is constant. It acts as an extension
+ * of the parent Frame type so that an OffsetFrame<Body>, for example,
+ * can be treated identically to a Body. This enables Frames to be filtered by
+ * their type (e.g. Physcial or not) and not whether or not it is an Offset.
+ *
+ * OffsetFrames also have the property that if they form a chain or a tree,
+ * each OffsetFrame shares the same Base which is the parent of the first/root
+ * OffsetFrame in the tree. This allows Solvers and algorithms to work directly
+ * with the Base which can be more efficient.
+ *
+ * @author Matt DeMers
+ * @author Ajay Seth
+ */
+template <class C = Frame>
```

Comment:
> I don't know what the template variable names mean

Should at least use doxygen `@tparam` to document them like other parameters.
It is conventional, but certainly not necessary, to use one-letter names for template arguments I think because they often get referenced on practically every line. Here a nicer name could be used I think. The other convention is to use a short name but provide a nice typedef. For example, the `std::` container classes are like `std::vector<T>` but then they define `typedef T element_type`, etc.